apps/site/lib/site_web/channels/vehicle_channel.ex

```diff
@@ -42,6 +45,14 @@ defmodule SiteWeb.VehicleChannel do
     stop_name = get_stop_name(vehicle.stop_id)
     trip = Schedules.Repo.trip(vehicle.trip_id)
 
+    # Sometimes trip is nil, so to avoid errors, we create a "polished" trip:
+    preprocessed_trip =
+      if trip == nil do
```

Comment:
This is totally fine, but just to mention another option: `nil` is falsy, so you can also do something like `trip || %{shape_id: nil}`.